main.py

```diff
@@ -0,0 +1,233 @@
+        return -1  #Negative
+    elif compound_score >= 0.25:
+        return 1   #Positive
+    else:
+        return 0   #Neutral
+
+def map_to_sentiment_category2(score: int):
+    if score in [1, 2, 3]:
+        return -1  #Negative
+    elif score in [4, 5]:
+        return 1   #Positive
+
+def map_compound_to_sentiment_category2(compound_score: float):
+    if compound_score < 0:
+        return -1  #Negative
+    else:
+        return 1   #Positive
+
+#Creating csv file to compare predictions with actual data
+def create_csv(data: pd.DataFrame, size: int):
+    if size == 3:
+        data['actual_sentiment'] = data['Score'].apply(map_to_sentiment_category3)
+        data['predicted_sentiment'] = data['compound'].apply(map_compound_to_sentiment_category3)
+    elif size == 2:
+        data['actual_sentiment'] = data['Score'].apply(map_to_sentiment_category2)
+        data['predicted_sentiment'] = data['compound'].apply(map_compound_to_sentiment_category2)
+    #Create final csv and save
+    result_data = data[['actual_sentiment', 'predicted_sentiment']]
+    result_data.to_csv(f'./sentiment_results{size}.csv', index=False)
+
+    return result_data
+
+def calculate_precision(actual_sentiment, predicted_sentiment, label: int):
+    TP = sum((actual == label) and (predicted == label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+    FP = sum((actual != label) and (predicted == label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+
+    precision = TP / (TP + FP) if TP + FP != 0 else 0.0
+    return precision
+
+def calculate_recall(actual_sentiment: pd.DataFrame, predicted_sentiment: pd.DataFrame, label: int):
+    TP = sum((actual == label) and (predicted == label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+    FN = sum((actual == label) and (predicted != label) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+
+    recall = TP / (TP + FN) if TP + FN != 0 else 0.0
+    return recall
+
+def calculate_accuracy(actual_sentiment: pd.DataFrame, predicted_sentiment: pd.DataFrame):
+    TP = sum((actual == predicted) for actual, predicted in zip(actual_sentiment, predicted_sentiment))
+    total_samples = len(actual_sentiment)
+    accuracy = TP / total_samples
+    return accuracy
+
+#Printing Confusion Matrix, Accurancy, Precision, Recall, F1-measure
+def calculate_metrics(size = 3): #size is number of Class Classification. Have to be > 1 and suitable for your data.
+    df = pd.read_csv(f'./sentiment_results{size}.csv')
+
+    actual_sentiment = df['actual_sentiment']
+    predicted_sentiment = df['predicted_sentiment']
+
+    print(f"METRICS FOR {size} CLASSES", end = "\n\n")
+
+    #Confusion matrix
+    if size == 3:
+        unique_labels = [-1, 0, 1]
+        confusion_matrix = np.zeros((3, 3), dtype=int)
+    elif size == 2:
+        unique_labels = [-1, 1]
+        confusion_matrix = np.zeros((2, 2), dtype=int)
+
```

Comment:
```suggestion
else:
    raise ValueError("Size can be only 2 or 3")
```

Otherwise you use `confusion_matrix` below but never initialize it.